### PR TITLE
WIP:Bugfix for issue 1734

### PR DESCRIPTION
**Pull request by TanayGahlot**

### CODE REVIEW

File: mne/evoked.py [hunk 1]
```diff
@@ -434,7 +434,7 @@ def plot_image(self, picks=None, exclude='bads', unit=True, show=True,
     def plot_topomap(self, times=None, ch_type='mag', layout=None, vmin=None,
                      vmax=None, cmap='RdBu_r', sensors=True, colorbar=True,
                      scale=None, scale_time=1e3, unit=None, res=64, size=1,
-                     format="%3.1f", time_format='%01d ms', proj=False,
+                     fmt="%3.1f", time_format='%01d ms', proj=False,
```
Comment (mainakjas) [inline mne/evoked.py]: we can't do this without a proper deprecation warning. Users are still using this

File: mne/io/base.py [hunk 2]
```diff
@@ -1412,7 +1412,7 @@ def n_times(self):
 
 ###############################################################################
 # Writing
-def _write_raw(fname, raw, info, picks, format, data_type, reset_range, start,
+def _write_raw(fname, raw, info, picks, fmt, data_type, reset_range, start,
```
Comment (mainakjas) [inline mne/io/base.py]: but it's fine here because it's private

File: mne/source_estimate.py [hunk 3]
```diff
@@ -2252,7 +2252,7 @@ def compute_morph_matrix(subject_from, subject_to, vertices_from, vertices_to,
     elif isinstance(morpher[1], list):
         morpher = morpher[0]
     else:
-        morpher = sparse_block_diag(morpher, format='csr')
+        morpher = sparse_block_diag(morpher, fmt='csr')
```
Comment (mainakjas) [inline mne/source_estimate.py]: I think this makes the test fail if scipy is not present. Do you know how to run tests locally?

Comment (TanayGahlot) [inline mne/source_estimate.py]: By using make command i guess ?

File: mne/viz/topomap.py [hunk 4]
```diff
@@ -1037,7 +1037,7 @@ def plot_evoked_topomap(evoked, times=None, ch_type='mag', layout=None,
     if colorbar:
         cax = plt.subplot(1, n + 1, n + 1)
         plt.colorbar(images[-1], ax=cax, cax=cax, ticks=[vmin, 0, vmax],
-                     format=format)
+                     format=fmt)
```
Comment (agramfort) [inline mne/viz/topomap.py]: @dengemann  matplotlib and scipy.sparse seem to be using format at different places.
We're not that odd using it in mne so I wonder...

Comment (dengemann) [inline mne/viz/topomap.py]: we cananot fix their issues. we can keep it clean for our code. it's certainly not better just because they do it. you just don't want to overwrite builtins.

Comment (agramfort) [inline mne/viz/topomap.py]: ok but it's a method of str type so there is not really any conflict

Comment (dengemann) [inline mne/viz/topomap.py]: Alex, `format` is an inbuilt function.

2015-02-15 18:03 GMT+01:00 Alexandre Gramfort <notifications@github.com>:

Comment (dengemann) [inline mne/viz/topomap.py]: https://docs.python.org/2/library/functions.html#format

2015-02-15 18:04 GMT+01:00 Denis-Alexander Engemann <
denis.engemann@gmail.com>:

Comment (agramfort) [inline mne/viz/topomap.py]: ok you win :)

File: mne/evoked.py (hunk 1 above)
Comment (TanayGahlot) [inline mne/evoked.py]: how do you suggest we do it? shall i revert the changes?

Comment (agramfort) [inline mne/evoked.py]: revert or better deprecate.

git grep deprecated

will help.

File: mne/source_estimate.py (hunk 3 above)
Comment (mainakjas) [inline mne/source_estimate.py]: yes, or use nosetests. Ask for help if you need. And make sure to read contributing guidelines: http://martinos.org/mne/stable/contributing.html

File: mne/evoked.py [hunk 5]
```diff
@@ -526,6 +526,12 @@ def plot_topomap(self, times=None, ch_type='mag', layout=None, vmin=None,
             starts 5 ms before and ends 5 ms after a given time point.
             Defaults to None, which means no averaging.
         """
+        if format is not None:
```
Comment (TanayGahlot) [inline mne/evoked.py]: Is this what you meant @agramfort

File: mne/fixes.py [hunk 6]
```diff
@@ -599,7 +599,7 @@ def _assert_is_not(expr1, expr2, msg=None):
     assert_is_not = _assert_is_not
 
 
-def _sparse_block_diag(mats, format=None, dtype=None):
+def _sparse_block_diag(mats, fmt=None, dtype=None):
```
Comment (agramfort) [inline mne/fixes.py]: is this the signature of scipy? as this function is a backward compat fix.

File: mne/evoked.py (hunk 5 above)
Comment (agramfort) [inline mne/evoked.py]: yes except you did not add the fmt param...

Comment (mainakjas) [inline mne/evoked.py]: no, but you didn't introduce the new argument `fmt`. You need to introduce it and then deprecate `format`

Comment (TanayGahlot) [inline mne/evoked.py]: how do i introduce it? can you give me an example?

Comment (mainakjas) [inline mne/evoked.py]: take a look here: https://github.com/mne-tools/mne-python/pull/1523/

Comment (TanayGahlot) [inline mne/evoked.py]: okay got it :-)

**Comment by TanayGahlot** [comment]
any clue on travis failure?

File: mne/io/base.py [hunk 7]
```diff
@@ -685,6 +685,16 @@ def save(self, fname, picks=None, tmin=0, tmax=None, buffer_size_sec=10,
...
+        format : str
+            Format to use to save raw data. Valid options are 'double',
+            'single', 'int', and 'short' for 64- or 32-bit float, or 32- or
+            16-bit integers, respectively. It is STRONGLY recommended to use
+            'single', as this is backward-compatible, and is standard for
+            maintaining precision. Note that using 'short' or 'int' may result
+            in loss of precision, complex data cannot be saved as 'short',
+            and neither complex data types nor real data stored as 'double'
+            can be loaded with the MNE command-line tools. See raw.orig_format
+            to determine the format the original data were stored in.
```
Comment (dengemann) [inline mne/io/base.py]: I think we should remove the doc for the old parameter ....

**Comment by dengemann** [comment]
@TanayGahlot from the Travis log

```
The command "nosetests --with-timer --timer-top-n 30 --verbosity=2 $COVERAGE" exited with 0.
3.12s$ if [ "${DEPS}" == "nodata" ]; then make flake; fi;
Running flake8
mne/viz/topomap.py:780:9: F841 local variable 'fmt' is assigned to but never used
mne/io/base.py:697:70: W291 trailing whitespace
mne/io/base.py:722:80: E501 line too long (83 > 79 characters)
mne/io/base.py:723:23: E128 continuation line under-indented for visual indent
mne/io/base.py:724:23: E128 continuation line under-indented for visual indent
5
make: *** [flake] Error 1
```

you have a style issue, Travis crashes if he find's pep8 violations

File: mne/io/base.py (hunk 7 above)
Comment (TanayGahlot) [inline mne/io/base.py]: okay cool

**Comment by TanayGahlot** [comment]
thanks @dengemann  i wl rectify that

**Comment by TanayGahlot** [comment]
well, format was never used in topomap.py, therefore fmt will also not be used. what to do now? travis is failing coz of that

**Comment by Eric89GXL** [comment]
for now you can add a `#  noqa` at the end of that line. It's weird we had a `kwarg` we didn't use, but I guess we could keep it for future compatibility.

**Comment by Eric89GXL** [comment]
Branch appears to no longer exist remotely, I'll take this over
